flake-modules/github-releases.nix

```
@@ -0,0 +1,96 @@
+{ config, lib, withSystem, ... }:
+{
+  options =
+    let
+      inherit (lib) mkOption mkOptionType types;
+    in
+    {
+      hercules-ci.github-releases = {
+        condition = lib.mkOption {
+          type = types.functionTo types.bool;
+          description = ''
+            Condition under which a release is going to be pushed.
+            This is a function accepting [HerculesCI parameters](https://docs.hercules-ci.com/hercules-ci-agent/evaluation#params-herculesCI)
+            and returning boolean.
+            By default, pushing happens if a tag is present.
+          '';
+          default = { tag, ... }: tag != null;
+          defaultText = ''
+            { tag, ... }: tag != null
+          '';
+        };
+        releaseTag = lib.mkOption {
+          type = types.functionTo types.str;
+          description = ''
+            Tag to be assigned to the release.
+          '';
+          default = herculesCI: herculesCI.config.repo.tag;
+          defaultText = "herculesCI: herculesCI.config.repo.tag";
+        };
+        files = lib.mkOption {
+          type = types.listOf types.str;
+          description = ''
+            List of asset _files_ --- no directories allowed.
+            Each path is in a form of `(/path/to/file|glob)[#display_label]`.
+          '';
+        };
```

```suggestion
          default = [ ];
        };
```

Make it so that the auto-enable works for repos that don't have or need a releases config.

---

Mmm I don't really understand you. There is no auto-enable, you must manually specify list of files

---

The thing is that your `enable` will be evaluated regardless of whether a flake wants to use releases or not. For those flakes, it needs a `default`.

> There is no auto-enable, you must manually specify list of files

:+1: That's what I meant with auto-enable: you only need to add files, not `enable = true;`. This is good.